test/e2e/network_test.go

```diff
@@ -301,13 +302,20 @@ func (tc *testContext) testEastWestNetworking(t *testing.T) {
 
 	linuxServerDeployment, err := tc.deployLinuxWebServer()
 	require.NoError(t, err)
+	// best effort to collect Linux web server logs before deleting it
+	defer func(tc *testContext, deployment *appsv1.Deployment) {
```

Comment:
Why does this need these intermediary arguments, rather than just calling `tc.collectDeploymentLogs(linuxServerDeployment)`? Is there a risk of the pointer being evaluated to the wrong value? If so that would effect `defer tc.deleteDeployment(linuxServerDeployment.GetName())` below as well.

Along with that, no need to have two separate defers called directly after each other, please move `tc.deleteDeployment(linuxServerDeployment.GetName())` into this one. That removes any question of the execution order of the defers

Reply:
> Why does this need these intermediary arguments, rather than just calling tc.collectDeploymentLogs(linuxServerDeployment)

Moved the to a func to wrap the error handling in a closure and ensure the params passed the order

> Is there a risk of the pointer being evaluated to the wrong value?
I don't anticipate any.


> If so that would effect defer tc.deleteDeployment(linuxServerDeployment.GetName()) below as well.
Yes, makes sense. I refactored to block to account of the correct order of cleanup steps.